sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/storage/PreferenceTokenStorage.java

```diff
@@ -0,0 +1,83 @@
+package com.microsoft.appcenter.identity.storage;
+
+import android.content.Context;
+import android.support.annotation.VisibleForTesting;
+
+import com.microsoft.appcenter.utils.context.AuthTokenContext;
+import com.microsoft.appcenter.utils.crypto.CryptoUtils;
+import com.microsoft.appcenter.utils.storage.SharedPreferencesManager;
+
+/**
+ * Storage for tokens that uses {@link SharedPreferencesManager}. Handles saving and encryption.
+ */
+public class PreferenceTokenStorage implements AuthTokenStorage {
+
+    /**
+     * {@link Context} instance.
+     */
+    private final Context mContext;
+
+    /**
+     * Default constructor.
+     *
+     * @param context {@link Context} instance.
+     */
+    PreferenceTokenStorage(Context context) {
+        mContext = context;
```

Comment:
mContext = context.getApplicationContext()

Reply:
@guperrot this is already done in CryptoUtils so it will crash if do this twice

Reply:
oh, weird I believe it should just return self if already app context

Reply:
it works in sasquatch, I just tested